CoreScriptsRoot/Modules/Settings/SettingsHub.lua

```diff
@@ -49,1155 +61,1275 @@ local resetButtonFlagSuccess, resetButtonFlagValue = pcall(function() return set
+
+        local hotKeyFunc = function(contextName, inputState, inputObject)
+            if inputState == Enum.UserInputState.Begin then
+                clickFunc()
+            end
+        end
 
-  local function createGui()
-    local PageViewSizeReducer = 0
-    if isSmallTouchScreen then
-      PageViewSizeReducer = 5
+        local hotKeyTable = {hotKeyFunc, hotkeys}
+        this.BottomBarButtons[#this.BottomBarButtons + 1] = {buttonName, hotKeyTable}
     end
 
-    this.ClippingShield = utility:Create'Frame'
-    {
-      Name = "SettingsShield",
-      Size = SETTINGS_SHIELD_SIZE,
-      Position = SETTINGS_SHIELD_ACTIVE_POSITION,
-      BorderSizePixel = 0,
-      ClipsDescendants = true,
-      BackgroundTransparency = 1,
-      Visible = true,
-      ZIndex = SETTINGS_BASE_ZINDEX,
-      Parent = RobloxGui
-    };
-
-    this.Shield = utility:Create'Frame'
-    {
-      Name = "SettingsShield",
-      Size = UDim2.new(1,0,1,0),
-      Position = SETTINGS_SHIELD_INACTIVE_POSITION,
-      BackgroundTransparency = SETTINGS_SHIELD_TRANSPARENCY,
-      BackgroundColor3 = SETTINGS_SHIELD_COLOR,
-      BorderSizePixel = 0,
-      Visible = false,
-      Active = true,
-      ZIndex = SETTINGS_BASE_ZINDEX,
-      Parent = this.ClippingShield
-    };
-    this.VRShield = utility:Create("Frame") {
-      Name = "VRBackground",
-      Parent = this.Shield,
-
-      BackgroundColor3 = SETTINGS_SHIELD_COLOR,
-      BackgroundTransparency = SETTINGS_SHIELD_TRANSPARENCY,
-      Position = UDim2.new(0, -4, 0, 24),
-      Size = UDim2.new(1, 8, 1, -40),
-      BorderSizePixel = 0,
-
-      Visible = false
-    }
-
-    this.Modal = utility:Create'TextButton' -- Force unlocks the mouse, really need a way to do this via UIS
-    {
-      Name = 'Modal',
-      BackgroundTransparency = 1,
-      Position = UDim2.new(0, 0, 1, -1),
-      Size = UDim2.new(1, 0, 1, 0),
-      Modal = true,
-      Text = '',
-      Parent = this.Shield,
-      Selectable = false
-    }
-
-    this.HubBar = utility:Create'ImageLabel'
-    {
-      Name = "HubBar",
-      ZIndex = this.Shield.ZIndex + 1,
-      BorderSizePixel = 0,
-      BackgroundColor3 = Color3.new(78/255, 84/255, 96/255),
-      BackgroundTransparency = 1,
-      Image = "rbxasset://textures/ui/Settings/MenuBarAssets/MenuBackground.png",
-      ScaleType = Enum.ScaleType.Slice,
-      SliceCenter = Rect.new(4,4,6,6),
-      Parent = this.Shield
-    };
-
-    local barHeight = 60
-    if isSmallTouchScreen then
-      barHeight = 40
-      this.HubBar.Size = UDim2.new(1,-10,0,40)
-      this.HubBar.Position = UDim2.new(0,5,0,6)
-    elseif isTenFootInterface then
-      barHeight = 100
-      this.HubBar.Size = UDim2.new(0,1200,0,100)
-      this.HubBar.Position = UDim2.new(0.5,-600,0.1,0)
-    else
-      this.HubBar.Size = UDim2.new(0,800,0,60)
-      this.HubBar.Position = UDim2.new(0.5,-400,0.1,0)
+    local function shouldShowBumperHints(lastInputType)
+        local gamepadInputTypes = {
```

Comment:
Maybe move this up and out so we can use it in many funcs? Perhaps this deserves a spot in a Utility area.